### ISSUE TITLE
Need upgrade path doc

### ISSUE THREAD

**Issue by BramGruneir** [cockroachdb/docs#7433]
We're missing a clear and concise doc that's explicit on exactly which versions need to be walked through to upgrade.

Our mixing of major and minor version numbers is already confusing for just about everyone, so lacking this doc makes it worse.

I'm thinking something along the lines of:

```
2.0.x -> 2.1.x
2.1.x -> 19.1.x
19.1.x -> 19.2.x
19.2.x -> 20.1.x
20.1.x -> 20.2.x
```

With some language about:

- what we consider a major version
- how we don't have minor versions 
- how patch versions can be applied
- how we must upgrade through each major version
- when can you downgrade (put a previous patch version on)
- how we finalize the version (and downgrades cannot happen afterwards)
- and how to prevent the finalization step if needed, but a warning about not doing so
- maybe a note about our release cadence and numbering because it's confusing.

**Comment by florence-crl** [comment]
Please include some clear answer to customer questions, such as:
> If possible I would like to know if we have to update from 19.1.x to 19.2.0 or if we can update directly to 19.2.7? Is there a recommendation around this?

**Comment by jseldess** [comment]
Good point. That should be clear on the how-to upgrade page, but it's not.
